test/jdk/tools/jlink/plugins/CompressorPluginTest.java

```diff
@@ -146,6 +146,65 @@ public void test() throws Exception {
                 new ResourceDecompressorFactory[]{
                     new ZipDecompressorFactory(),
                 }, Collections.singletonList(".*Exception.class"));
+
+        // compress level zip-0 == no compression
+        Properties optionsZip0 = new Properties();
+        DefaultCompressPlugin compressPluginZip0 = new DefaultCompressPlugin();
+        options0.setProperty(compressPluginZip0.getName(), "zip-0");
+        checkCompress(classes, compressPlugin,
+                optionsZip0,
+                new ResourceDecompressorFactory[]{
+                });
+
+        // compress level zip-[1-9] == varied compression levels
+        for(int i = 1; i < 10; i++) {
+            Properties optionsZip = new Properties();
+            compressPlugin = new DefaultCompressPlugin();
+            optionsZip.setProperty(compressPlugin.getName(), "zip-" + i);
+            checkCompress(classes, compressPlugin,
+                    optionsZip,
+                    new ResourceDecompressorFactory[]{
+                            new ZipDecompressorFactory(),
+                    });
+        }
+
+        // compress level zip-[1-9] == varied compression levels + filter
+        for(int i = 1; i < 10; i++) {
+            Properties optionsZip = new Properties();
+            compressPlugin = new DefaultCompressPlugin();
+            optionsZip.setProperty(DefaultCompressPlugin.FILTER, "**Exception.class");
+            optionsZip.setProperty(compressPlugin.getName(), "zip-" + i);
+            checkCompress(classes, compressPlugin,
+                    optionsZip,
+                    new ResourceDecompressorFactory[]{
+                            new ZipDecompressorFactory(),
+                    }, Collections.singletonList(".*Exception.class"));
+        }
+
+        testBadCompressProps(classes, "zip-10");
+        testBadCompressProps(classes, "zip-badarg");
+        testBadCompressProps(classes, "zip-10000000");
+
+    }
+
+    private void testBadCompressProps(ResourcePool classes, String compressArg) throws Exception {
+        Properties badProps = new Properties();
+        DefaultCompressPlugin compressPlugin = new DefaultCompressPlugin();
+        badProps.setProperty(compressPlugin.getName(), compressArg);
+        try {
+            checkCompress(classes, compressPlugin,
+                    badProps,
+                    new ResourceDecompressorFactory[]{
+                            new ZipDecompressorFactory(),
+                    });
+        } catch (IllegalArgumentException e) {
+            if (e.getMessage().contains("Invalid compression level")) {
+                return;
+            }
```

Comment:
Should be print the stacktrace if the message is not the one we expect, to help debug any failures? Or perhaps rethrow the exception to cause the test to fail?